src/AmplifyGeofenceControl/ui.ts

```diff
@@ -374,6 +374,7 @@ export function AmplifyGeofenceControlUI(
     checkbox.addEventListener("click", function () {
       if ((checkbox as HTMLInputElement).checked) {
         geofenceControl.displayGeofence(geofence.geofenceId);
+        geofenceControl.fitGeofence(geofence.geofenceId);
```

Comment:
Would it make sense to try to `fitAllGeofences` every time we check an ID? Does `this._loadedGeofences` only give us the "checked" geofences? I could see a situation where a user checks a large geofence which will fit the map view, then check a smaller geofence which will be fitted and cut off some of the larger geofence.

Reply:
Ok nice, `this._displayedGeofences` works! I do see a tradeoff here of needing to do a nested iteration for all of our displayed geofences every time we want to fit the zoom, which would be more expensive than just fitting for one geofence at a time.

But I think fitting for all displayed geofences is the better experience.